code/controllers/subsystem/monitor.dm

```diff
@@ -0,0 +1,125 @@
+SUBSYSTEM_DEF(monitor)
+	name = "Monitor"
+	init_order = INIT_ORDER_MONITOR
+	runlevels = RUNLEVEL_GAME
+	///The next world.time for wich the monitor subsystem refresh the state
+	var/scheduled = 0
+	///The current state
+	var/current_state = STATE_BALANCED
+	///The last state
+	var/last_state = STATE_BALANCED
+	///If we consider the state as a stalemate
+	var/stalemate = FALSE
+	///The current state points
+	var/current_points = 0
+	///The number of time we had the same state consecutively
+	var/stale_counter = 0
+	///The number of humans on ground
+	var/human_on_ground = 0
+	///The number of humans being in either lz1 or lz2
+	var/human_in_FOB = 0
+	///The number of time most of humans are in FOB consecutively
+	var/humans_all_in_FOB_counter = 0
+	///TRUE if we detect a state of FOB hugging
+	var/FOB_hugging = FALSE
+	
+
+
+/datum/controller/subsystem/monitor/Initialize(time, zlevel)
+	if(CONFIG_GET(flag/monitor_disallowed))
+		can_fire = 0
+	reschedule()
+	return ..()
+
+/datum/controller/subsystem/monitor/fire(resumed = 0)
+	if(!resumed)
+		check_state() //only check these if we aren't resuming a paused fire
+
+
+///checks if we should refresh the director state, and reschedules if necessary
+/datum/controller/subsystem/monitor/proc/check_state()
+	if(scheduled <= world.time)
+		etablish_state()
+		reschedule()
+
+///decides which world.time we should calculate a new monitor state at.
+/datum/controller/subsystem/monitor/proc/reschedule()
+	scheduled = world.time + FREQUENCY_STATE_CALCULATION
```

Comment:
Why not instead use the subsystems wait?

Reply:
I just copied a subsystem you made Tivi

Reply:
The events subsystem rescedules because events have variable trigger times while yours could readily use the wait variable